aws/resource_aws_apigatewayv2_deployment_test.go

```diff
@@ -0,0 +1,161 @@
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAWSAPIGatewayV2DeploymentExists(resourceName, &apiId, &v),
+					resource.TestCheckResourceAttr(resourceName, "auto_deployed", "false"),
+					resource.TestCheckResourceAttr(resourceName, "description", "Test description"),
+				),
+			},
+			{
+				ResourceName:      resourceName,
+				ImportStateIdFunc: testAccAWSAPIGatewayV2DeploymentImportStateIdFunc(resourceName),
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+			{
+				Config: testAccAWSAPIGatewayV2DeploymentConfig_basic(rName, "Test description updated"),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAWSAPIGatewayV2DeploymentExists(resourceName, &apiId, &v),
+					resource.TestCheckResourceAttr(resourceName, "auto_deployed", "false"),
+					resource.TestCheckResourceAttr(resourceName, "description", "Test description updated"),
+				),
+			},
+		},
+	})
+}
+
+func TestAccAWSAPIGatewayV2Deployment_disappears(t *testing.T) {
+	var apiId string
+	var v apigatewayv2.GetDeploymentOutput
+	resourceName := "aws_apigatewayv2_deployment.test"
+	rName := acctest.RandomWithPrefix("tf-acc-test")
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckAWSAPIGatewayV2DeploymentDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAWSAPIGatewayV2DeploymentConfig_basic(rName, "Test description"),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAWSAPIGatewayV2DeploymentExists(resourceName, &apiId, &v),
+					testAccCheckAWSAPIGatewayV2DeploymentDisappears(&apiId, &v),
+				),
+				ExpectNonEmptyPlan: true,
+			},
+		},
+	})
+}
+
+func testAccCheckAWSAPIGatewayV2DeploymentDestroy(s *terraform.State) error {
+	conn := testAccProvider.Meta().(*AWSClient).apigatewayv2conn
+
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "aws_apigatewayv2_deployment" {
+			continue
+		}
+
+		_, err := conn.GetDeployment(&apigatewayv2.GetDeploymentInput{
+			ApiId:        aws.String(rs.Primary.Attributes["api_id"]),
+			DeploymentId: aws.String(rs.Primary.ID),
+		})
+		if isAWSErr(err, apigatewayv2.ErrCodeNotFoundException, "") {
+			continue
+		}
+		if err != nil {
+			return err
+		}
+
+		return fmt.Errorf("API Gateway v2 deployment %s still exists", rs.Primary.ID)
+	}
+
+	return nil
+}
+
+func testAccCheckAWSAPIGatewayV2DeploymentDisappears(apiId *string, v *apigatewayv2.GetDeploymentOutput) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		conn := testAccProvider.Meta().(*AWSClient).apigatewayv2conn
+
+		_, err := conn.DeleteDeployment(&apigatewayv2.DeleteDeploymentInput{
+			ApiId:        apiId,
+			DeploymentId: v.DeploymentId,
+		})
+
+		return err
+	}
+}
+
+func testAccCheckAWSAPIGatewayV2DeploymentExists(n string, vApiId *string, v *apigatewayv2.GetDeploymentOutput) resource.TestCheckFunc {
```

Comment:
I like how in this and your other PRs you've added returning the ID to our existing pattern of returning the object.

Reply:
The returned objects don't include all the necessary information to be able to identify the resource which is different from other AWS APIs.

Reply:
Ah, I see. It's a minor convenience/cosmetic thing, but I've seen a number of use cases where it would be handy, or at least meaning fewer dereferences :)